climakitae/explore/amy.py

```diff
@@ -203,6 +296,574 @@ def _return_diurnal(y: xr.DataArray) -> xr.DataArray:
     return df_amy
 
 
+def get_climate_profile(**kwargs) -> pd.DataFrame:
+    """
+    High-level function to compute climate profiles using warming level data.
+
+    This function retrieves climate data and computes average meteorological year
+    profiles using the 8760 analysis approach. It combines data retrieval and
+    profile computation in a single call.
+
+    Parameters
+    ----------
+    **kwargs : dict
+        Keyword arguments for data selection. Allowed keys:
+        - variable (Optional) : str, default "Air Temperature at 2m"
+        - resolution (Optional) : str, default "3 km"
+        - warming_level (Required) : List[float], default [1.2]
+        - cached_area (Optional) : str or List[str]
+        - units (Optional) : str, default "degF"
+        - latitude (Optional) : float or tuple
+        - longitude (Optional) : float or tuple
+        - days_in_year (Optional) : int, default 365
+        - q (Optional) : float | list[float], default [], quantile for profile calculation
+        - no_delta (optional) : bool, default False, if True, do not apply baseline subtraction, return raw future profile
+
+    Returns
+    -------
+    pd.DataFrame
+        Average meteorological year table for each warming level, with days of year as
+        the index and hour of day as the columns. If multiple warming levels exist,
+        they will be included as additional column levels. Units and metadata are
+        preserved in the DataFrame's attrs dictionary.
+
+    Examples
+    --------
+    >>> profile = get_climate_profile(
+    ...     variable="Air Temperature at 2m",
+    ...     warming_level=[1.5, 2.0, 3.0],
+    ...     units="degF"
+    ... )
+
+    >>> profile = get_climate_profile(warming_level=[2.0])
+    """
+    # Extract parameters for compute_profile
+    days_in_year = kwargs.pop("days_in_year", 365)
+    q = kwargs.pop("q", 0.5)
+    no_delta = kwargs.get("no_delta", False)
+
+    # Retrieve the climate data
+    print("📊 Retrieving climate data...")
+    with tqdm(total=2, desc="Data retrieval", unit="dataset") as pbar:
+        historic_data, future_data = retrieve_profile_data(**kwargs)
+        pbar.update(2)
+
+    print(future_data.attrs)
+
+    # Call compute_profile with the processed data
+    # Compute profiles for both historical and future data
+    if isinstance(future_data, xr.Dataset):
+        var_name = list(future_data.data_vars.keys())[0]
+        future_profile_data = future_data[var_name]
+    else:
+        future_profile_data = future_data
+
+    if isinstance(historic_data, xr.Dataset):
+        var_name = list(historic_data.data_vars.keys())[0]
+        historic_profile_data = historic_data[var_name]
+    else:
+        historic_profile_data = historic_data
+
+    # Compute profiles for both datasets
+    print("⚙️  Computing climate profiles...")
+
+    with tqdm(total=2, desc="Profile computation", unit="profile") as pbar:
```

Comment:
I think if you have total=2 when there's only one object being computed (no_delta=False), then tqdm makes it seem like two datasets were still computed and each one taking half as long as the one dataset being computed actually took (i.e. future takes 100 seconds, tqdm shows 50 sec/dataset), which could be misleading to the user